### DISCUSSION TITLE
Reactive Support in Spring Pulsar

### DISCUSSION THREAD

**Discussion by onobc** [spring-projects/spring-pulsar#99] [General]
### Overview
This is a general place to discuss Reactive support in Spring Pulsar. Currently the discussion is scattered about several issues. Let's move the discussion(s) here instead.

**Comment by onobc** [comment]
The original discussion started [here](https://github.com/spring-projects-experimental/spring-pulsar/issues/7).

**Comment by sobychacko** [comment]
Hi, thanks, @onobc, for creating this discussion.

https://github.com/apache/pulsar-client-reactive is going to be the official Pulsar reactive client implementation. Spring for Apache Pulsar will consume this official implementation eventually to provide a Spring friendly reactive extension on top. 
cc @lhotari @migroskub

**Comment by code-magician323** [comment]
What are the abstractions that are planned to be on Spring Pulsar on this subject? I think that a `@ReactiveListener` & `ReactivePulsarTemplate#send` would be enough as a start since they will enable users to use the built-in Pulsar auto-configuration with reactive streams.

**Reply by onobc** [reply]
That sounds like a good start to me as well @code-magician323 . We will definitely not try to "boil the ocean" but rather provide a good starting place and learn from that and expand upon it.

**Reply by lhotari** [reply]
Makes sense to start with something simple. @cbornet is starting some experiments in integrating https://github.com/apache/pulsar-client-reactive with https://github.com/spring-projects-experimental/spring-pulsar .

**Comment by migroskub** [comment]
@onobc Is it reasonable to expect this to happen until GA?

**Reply by onobc** [reply]
Hi @migroskub , our goal is to have some support for reactive pulsar in by GA which we are planning to happen before Spring One (~2 months).

Thanks for your patience and staying in touch.  It is much appreciated. I will reach out asap once we have something to start looking at.  We will also ping you on the PRs as they come in as I know you are interested. 

Cheers

**Reply by migroskub** [reply]
Thanks! I'm glad to hear that the project is about to be talked about at Spring One!

**Comment by cbornet** [comment]
Documentation is not yet available but reactive support has been merged into master.
See `@ReactivePulsarListener` and `ReactivePulsarTemplate` for details.

**Comment by onobc** [comment]
Doing a bit of GH discussions housekeeping...

Since this has been implemented - going to close the discussion as resolved.

Thanks to all who participated in the discussion.
